mrp_restrict_lot/models/stock_rule.py

```diff
@@ -0,0 +1,37 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class StockRule(models.Model):
+    _inherit = "stock.rule"
+
+    def _prepare_mo_vals(
+        self,
+        product_id,
+        product_qty,
+        product_uom,
+        location_id,
+        name,
+        origin,
+        company_id,
+        values,
+        bom,
+    ):
+        vals = super()._prepare_mo_vals(
+            product_id,
+            product_qty,
+            product_uom,
+            location_id,
+            name,
+            origin,
+            company_id,
+            values,
+            bom,
+        )
+        lot_id = values.get("restrict_lot_id")
+        if lot_id:
+            vals["lot_producing_id"] = lot_id
```

Comment:
Just a question. "lot_producing_id" is a key used in another module or only here ?

Reply:
That's the lot field on MO, from main mrp module : https://github.com/odoo/odoo/blob/14.0/addons/mrp/models/mrp_production.py#L104